### ISSUE TITLE
Why attention_mask should repeat 'sp_size' here?

### ISSUE THREAD

**Issue by colian** [PKU-YuanGroup/Open-Sora-Plan#373]
Thanks for your great work!

https://github.com/PKU-YuanGroup/Open-Sora-Plan/blob/3c08cb2fef031dc659fbcfb33acac4c515d7bc51/opensora/utils/communications.py#L119

Since the input size of attention_mask  is (1, 24, 60, 80), why still to repeat sp_size times here?

**Comment by apprivoiser** [comment]
The reason for repeating sp_size times is that each device needs to obtain the complete mask to run FA(there are sp_size devices in a sequence parallel group) .

**Comment by colian** [comment]
I see..  Thanks for your fast reply. Could u share some materials that introduce this "sp" scheme for full attention? I don't quite understand that why separates frames into different gpu cards can achieve the full attention.

**Comment by LinB203** [comment]
你可以查看一下我们在计算attention时候的all2all操作。简单地说就是输入按照帧切分，但是算attention时候按照head切分。

**Comment by colian** [comment]
非常感谢！参考 DEEPSPEED ULYSSES 这篇论文，我看明白了。
